### DISCUSSION TITLE
Slice functionality - issue unable to use Spawn component

### DISCUSSION THREAD

**Discussion by soltsmank** [o3de/o3de#6638] [Help Wanted] [answered]
Hello O3DEeners!

I am new to O3DE. Trying to make a simple game, but run to an issue. 
Unable to locate nowhere "Create a slice" functionality in the editor, when right-clicking on Entity the only option available is "Create a prefab". Is it still not ported  from Lumberyard.

Slices required by Spawn component, when I add it.

**EDIT:**
Let me rephrase my question. For example, I am making Tetris game. (Have all assets of square, |,  L , + , and others)
At the top inserted Entity with Spawn component attached. Lets make it simple and spawn only | pipes. 
Here is the issue, **how to create a slice which required by spawn component?**

Did I miss to add a Gem on configuration ?
Looking for docs / examples for spawning Entities dynamically. 

Thanks.

**Comment by FiniteStateGit** [comment, marked as answer]
Hi @soltsmank,

Slices and the spawn components are deprecated. Instead, ensure you have the Prefab Builder gem enabled for your project.  Then create a prefab for each of the objects you want to spawn.  In Script Canvas, you can use the Spawn node (found under in the category Spawning *not* Gameplay).  In Node Inspector, select the prefab you created in the field m_spawnableAsset.

![spawn_node](https://user-images.githubusercontent.com/31811688/148587991-bed82fac-4657-4ad0-9d0f-9995bbc172c4.png)

**Reply by soltsmank** [reply]
Thanks @FiniteStateGit ,  I managed to procced now.

**Comment by soltsmank** [comment]
One more question on the same SPAWN topic.

Is there any option to provide m_spawnableAsset by input to 'Spawn' node?
Lets say I have several possible assets for spawn, example: box_1, box_2, box_3,..
I thought that I could use 'RandomNumber' node to concatenate with prefix of box_ and input it as an asset name to 'Spawn' node.

For now I see only option to instantiate 'Spawn' nodes for every spawnable asset and just call 'Random Signal' node to choose one of them.

Any other ideas? (Focus on script canvas only, no Lua)

**Reply by FiniteStateGit** [reply]
You are correct that you need to create spawn nodes for every asset and use something like 'Random Signal' to choose between them.  I wouldn't be surprised if the node was changed in the future to allow programmatic selection spawnable assets.

**Reply by soltsmank** [reply]
Thanks for confirmation.
